### DISCUSSION TITLE
[Bug]: Build fails in ci/cd conditions due to file-system-cache dependency

### DISCUSSION THREAD

**Discussion by xzilja** [storybookjs/storybook#25504] [Help] [answered]
### Describe the bug

Builds seem to fail in GitHub actions / Vercel while passing locally (via same node version, frozen lockfile). Issue seems to come from running `storybook build` command in ci and produces following error

```


node:internal/modules/cjs/loader:1078
--
09:08:06.785 | throw err;
09:08:06.785 | ^
09:08:06.785 |  
09:08:06.785 | Error: Cannot find module './util'
09:08:06.785 | Require stack:
09:08:06.787 | - /vercel/path0/node_modules/file-system-cache/lib/common/index.js
09:08:06.787 | - /vercel/path0/node_modules/file-system-cache/lib/FileSystemCache.js
09:08:06.787 | - /vercel/path0/node_modules/file-system-cache/lib/index.js
09:08:06.787 | - /vercel/path0/node_modules/@storybook/core-common/dist/index.js
09:08:06.787 | - /vercel/path0/node_modules/@storybook/telemetry/dist/index.js
09:08:06.787 | - /vercel/path0/node_modules/@storybook/cli/dist/generate.js
09:08:06.787 | - /vercel/path0/node_modules/@storybook/cli/bin/index.js
09:08:06.788 | - /vercel/path0/node_modules/storybook/index.js
09:08:06.788 | at Module._resolveFilename (node:internal/modules/cjs/loader:1075:15)
09:08:06.788 | at Module._load (node:internal/modules/cjs/loader:920:27)
09:08:06.788 | at Module.require (node:internal/modules/cjs/loader:1141:19)
09:08:06.788 | at require (node:internal/modules/cjs/helpers:110:18)
09:08:06.788 | at Object.<anonymous> (/vercel/path0/node_modules/file-system-cache/lib/common/index.js:30:27)
09:08:06.788 | at Module._compile (node:internal/modules/cjs/loader:1254:14)
09:08:06.788 | at Module._extensions..js (node:internal/modules/cjs/loader:1308:10)
09:08:06.788 | at Module.load (node:internal/modules/cjs/loader:1117:32)
09:08:06.788 | at Module._load (node:internal/modules/cjs/loader:958:12)
09:08:06.788 | at Module.require (node:internal/modules/cjs/loader:1141:19) {
09:08:06.788 | code: 'MODULE_NOT_FOUND',
09:08:06.788 | requireStack: [
09:08:06.788 | '/vercel/path0/node_modules/file-system-cache/lib/common/index.js',
09:08:06.788 | '/vercel/path0/node_modules/file-system-cache/lib/FileSystemCache.js',
09:08:06.789 | '/vercel/path0/node_modules/file-system-cache/lib/index.js',
09:08:06.789 | '/vercel/path0/node_modules/@storybook/core-common/dist/index.js',
09:08:06.789 | '/vercel/path0/node_modules/@storybook/telemetry/dist/index.js',
09:08:06.789 | '/vercel/path0/node_modules/@storybook/cli/dist/generate.js',
09:08:06.792 | '/vercel/path0/node_modules/@storybook/cli/bin/index.js',
09:08:06.792 | '/vercel/path0/node_modules/storybook/index.js'
09:08:06.792 | ]
09:08:06.792 | }
09:08:06.792 |  
09:08:06.793 | Node.js v18.15.0
```

Googling around, doesn't bring much up besides this being a telemetry issue, but even with 
```
core: {
    disableTelemetry: true
}
```

Issue still persists

### To Reproduce

Build storybook in ci

### System

```shell
Environment Info:

  System:
    OS: macOS 13.4.1
    CPU: (10) arm64 Apple M1 Pro
  Binaries:
    Node: 20.3.1 - /opt/homebrew/bin/node
    Yarn: 1.22.19 - /opt/homebrew/bin/yarn
    npm: 9.6.7 - /opt/homebrew/bin/npm
  Browsers:
    Chrome: 114.0.5735.198
    Firefox: 112.0.2
    Safari: 16.5.1
```

**Comment by xzilja** [comment]
Setting ` "file-system-cache": "2.0.0"` explicitly as dependency seems to solve the issue. By default it resolves to `2.4.0`. Similarly, latest 7.1 alpha also works out of the box, perhaps a fix was introduced there?

**Comment by koskinpark** [comment, marked as answer]
Add
```
  "resolutions": {
    "file-system-cache": "2.3.0"
  },
```
if you are using yarn.

For npm https://docs.npmjs.com/cli/v8/configuring-npm/package-json#overrides

It's a quick fix which you can use to control version of buggy package. It's a temporary solution. We need to wait for the fix of it in original https://github.com/philcockfield/file-system-cache repo

**Comment by twigy11** [comment]
I have the same problem from today compilation

the workarrounde works fine

`  "overrides": {
    "file-system-cache": "2.3.0"
  },`

**Comment by y-a-v-a** [comment]
`file-system-cache` is explicitly dependent on typescript@5 per the exact version that Storybook is depending on (being `2.3.0`) – see this commit: https://github.com/philcockfield/file-system-cache/commit/743f0c17b2280b49f5d0e0cae1440181b489588c

`file-system-cache` depends on `@types/ramda` which has a carret dependency on `types-ramda` that just [last week ](https://github.com/ramda/types/releases/tag/v0.29.5) updated their types using Typescript 5 specific features.

Storybook depends on typescript@4.9.x.
This broke our build.

Our solution was indeed to use `resultions`, but in this case not for `file-system-cache` but specifically for `types-ramda`.

```json
"resolutions": {
  "types-ramda": "0.29.4",
}
```

Maybe someone benefits from this analysis.
